onert-micro/eval-driver/TrainingDriver.cpp

```diff
@@ -100,118 +93,264 @@ int entry(int argc, char **argv)
+  config.wof_ptr = wof_data.data();
+  config.train_mode = true;
   {
-    auto input_data = reinterpret_cast<char *>(interpreter.allocateInputTensor(0));
+    onert_micro::OMTrainingConfig trainConfig;
+    trainConfig.lambda = lambda;
+    trainConfig.optimization_strategy = onert_micro::SGD;
+    trainConfig.beta_squares = 0.999f;
+    trainConfig.beta = 0.9f;
+    trainConfig.batches = 1;
+
+    config.train_config = trainConfig;
+  }
 
-    std::memcpy(input_data, test_data_u8, size);
+  // Create training interpreter and import models
+  onert_micro::OMTrainingInterpreter train_interpreter;
+  train_interpreter.import(circle_model.data(), backprop_model.data(), config);
 
-    interpreter.interpret();
-    auto data = reinterpret_cast<float *>(interpreter.readOutputTensor(0));
+  const auto model_inputs_num = train_interpreter.getNumberOfInputs();
+  const auto model_outputs_num = train_interpreter.getNumberOfOutputs();
+  const auto model_targets_num = train_interpreter.getNumberOfTargets();
 
-    printf("Sample № = %d\n", i);
-    for (int j = 0; j < output_size / sizeof(float); ++j)
-    {
-      printf("j = %d: predicted_result = %f, correct_result = %f\n", j, data[j],
-             reinterpret_cast<float *>(label_test_data)[j + i * output_size / sizeof(float)]);
-    }
-    printf("\n");
-    test_data_u8 += size;
-  }
+  if (model_outputs_num != model_targets_num)
+    throw std::runtime_error("Output model size and target size have to be equal");
 
-  float mse_result = 0.0f;
+  assert(model_inputs_num == 1);
+  assert(model_outputs_num == 1);
+  assert(model_targets_num == 1);
 
-  settings.metric = luci_interpreter::training::MSE;
-  onert_micro_training.test(num_train_data_samples, reinterpret_cast<const uint8_t *>(train_data),
-                            reinterpret_cast<const uint8_t *>(label_train_data),
-                            reinterpret_cast<void *>(&mse_result));
+  // Save model input/output/target size
+  const auto input_size = train_interpreter.getInputSizeAt(0);
+  const auto output_size = train_interpreter.getOutputSizeAt(0);
+  const auto target_size = train_interpreter.getTargetSizeAt(0);
 
-  float mae_result = 0.0f;
+  // train data
+  std::unique_ptr<float []> train_input_data(new float [input_size]);
+  std::unique_ptr<float []> train_target_data(new float [target_size]);
 
-  settings.metric = luci_interpreter::training::MAE;
-  onert_micro_training.test(num_train_data_samples, reinterpret_cast<const uint8_t *>(train_data),
-                            reinterpret_cast<const uint8_t *>(label_train_data),
-                            reinterpret_cast<void *>(&mae_result));
+  // Test data
+  std::unique_ptr<float []> test_input_data(new float [input_size]);
+  std::unique_ptr<float []> test_target_data(new float [target_size]);
 
-  printf("MSE_ERROR TRAIN = %f\n", mse_result);
+  for (uint32_t e = 0; e < training_epochs; ++e)
+  {
+    train_interpreter.set_training_mode(true);
```

Comment:
@BalyshevArtem 

Does this code support transfer learning?

Reply:
Do you mean by transfer learning that only certain last layers are trained? If yes, then TrainingConfigure tool is responsible for how many and which specific layers are trained in this draft for `proposal_1` from https://github.com/Samsung/ONE/issues/12873#issuecomment-2059003990. 
from TrainingDriver we cannot change and choose which layers will be trained in this variant.

Reply:
I understood that the backprop graph contains which layers will be trained, and TrainingDriver cannot change the training layer information other than turning training on and off. I'm trying to add `trainable` property in circle+ file format. So I wonder how you handle the trasfer learning feature. Thank you for your explanation.